testsupport/common/src/main/java/com/adobe/testing/s3mock/testsupport/common/S3MockStarter.java

```diff
@@ -211,6 +219,11 @@ public void checkServerTrusted(final X509Certificate[] arg0, final String arg1,
       return this;
     }
 
+    public BaseBuilder<T> withSecureConnection(final boolean secureConnection) {
+      arguments.put(S3MockApplication.PROP_SECURE_CONNECTION, secureConnection);
```

Comment:
Unfortunately the way everything is setup the `arguments` map is being passed through to the constructor of various classes in the abstract `build` method. I see a couple of options we could do:
  1. Use the single Map for both server and client configuration (currently implemented)
  2. Create two maps -- one to hold the server configuration, the other to hold the client configuration
  3. Pass the Builder class through to the various constructors which would allow getter access server properties and client properties.

Reply:
Ah, right. I'm not totally psyched about the current approach we're using, but changing that would probably break the API of the S3MockStarter... That's something for a later version.

Then let's do it how you suggested it in the first place... 👍